Migração controllers - código limpo

ocaiooo opened the pull request.

ocaiooo committed:
Alterações código

ocaiooo committed:
Update Estadios.cs

ocaiooo committed:
Merge branch 'master' into dev

ocaiooo committed:
Exclusão EstadioController

ocaiooo committed:
Separação dos controllers

ocaiooo committed:
Inicio da clean

ocaiooo committed:
minimal changes

ocaiooo committed:
GetsById all entities

ocaiooo committed:
get using generic classes

ocaiooo committed:
Migração controllers

tiags38 commented on TimesBD/Controllers/CompradorController.cs:
usar a classe Result para retorno da API

tiags38 commented on TimesBD/Controllers/EstadioController.cs:
usar a classe Result para retorno da API

tiags38 commented on TimesBD/Repositories/ApiRep.cs:
usar a classe result na ApiRep também 
criar novo campo na Result : public string JsonResponse {get;set;}

tiags38 commented on TimesBD/Controllers/TimeDbControllerBase.cs:
instanciar a businessclass como protected e usar apartir daqui

tiags38 commented on TimesBD/Business/BusinessClass.cs:
```diff
@@ -0,0 +1,521 @@
+        return await _sqlRep.GetQueryAsync<T>(sql);
+    }
+    
+    public async Task<Result> DeleteEntityAsync<T>(int id) where T : class
```
pegar o numero de linhas afetadas para validar se algyuem foi excluido

tiags38 commented on TimesBD/Business/BusinessClass.cs:
```diff
@@ -0,0 +1,521 @@
+        await DeleteEntityAsync<Venda>(id);
+    }
+    
+    public async Task<IEnumerable<T>> GetEntityByIdAsync<T>(int? id) where T : class
```
adicionar o retorno Result, alem do ienumerable para verificar se deu erro ou sucesso

tiags38 commented on TimesBD/Controllers/TimeDbControllerBase.cs:
```diff
@@ -0,0 +1,19 @@
+
+namespace TimesBD.Controllers
+{
+    public class TimeDbControllerBase : ControllerBase
```
instanciar business aqui

ocaiooo committed:
Refactor and simplify business logic and API controllers

The business logic functions were refactored to simplify the validation process, improve error handling, and optimize SQL operations. Additionally, the related API controllers' responses were enhanced to be more informative. Someone using the API can now understand what's happening clearer with these responses.

ocaiooo committed:
SPROC
